src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -285,5 +282,49 @@ ExpressionSyntax initializer
 			.GetDeclarationSyntax<SyntaxNode>()
 			.FirstAncestorOrSelf<MemberDeclarationSyntax>()
 			.GetLocation();
+
+		private Location GetLocationOfBaseClass( ITypeSymbol type ) {
+			// Consider the following valid code:
+			//
+			// class Base {}
+			// interface IA { }
+			// interface IB { }
+			//
+			// partial class Foo { }
+			// partial class Foo : Base { }
+			// partial class Foo : Base, IA { }
+			// partial class Foo : IB { }
+			//
+			// Our goal is to get one of the mentions of Base after "Foo :" to
+			// place the diagnostic. There may be multiple, but for valid code
+			// it will be the first type in the list for one of the decls.
+			// (interfaces always go after the base class, if it is listed.)
+
+			var candidates = type.DeclaringSyntaxReferences
+				.Select( r => r.GetSyntax() )
+				.Cast<ClassDeclarationSyntax>()
+				// Take _at most_ the first item from each BaseList.Types
+				.SelectMany( r => r.BaseList.Types.Take( 1 ) );

```

Comment:
Even though ` Foo: IA, Base` won't compile, our analyzer may still run?

🤷

Reply:
Yep, so my thought is that the semantic model already told us that our base type is something other than `System.Object`, so it seems likely that `Foo : IA, Base` either isn't there or there is another `Foo : Base` decl.

That may not be true, so I added a fallback.